src/asami/query.cljc

```diff
@@ -638,27 +638,52 @@
       ;; calculate the aggregates from the final results and project
       (aggregate-over find inner-results))))
 
-(s/defn rewrite-wildcards
-  "In the :where sequence of query replace all occurrences of _ with
-  unique variables."
-  [{:keys [where] :as query}]
+
+(defn ^:private fresh []
+  (gensym "?__"))
+
+(defn ^:private map-epv
+  "In the :where sequence of query apply f to each EPV pattern."
+  [f {:keys [where] :as query}]
   (assoc query
          :where (map (fn [clause]
```

Comment:
I'd rename this. `clause` actually refers to the entire `:where` expression. In this case the parameter is a `constraint` (`:where` clauses are a seq of constraints)

Reply:
Thanks for correcting this. 👍